dashboard/app/models/user_level.rb

```diff
@@ -138,15 +130,30 @@ def before_unsubmit
     end
   end
 
-  def has_autolocked?(stage)
-    return false unless stage.lockable?
-    unlocked_at && unlocked_at < AUTOLOCK_PERIOD.ago
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`, so the following
+  # are its getter and setter
+  def locked
+    unlocked_at.nil? || unlocked_at < AUTOLOCK_PERIOD.ago
   end
 
-  def locked?(stage)
+  def locked=(is_locked)
+    self.unlocked_at = is_locked ? nil : Time.now
+  end
+
+  # this is the "locked" value we return to the client. if the lesson isn't
+  # lockable or we are an authorized teacher, we always return `false`.
+  def show_as_locked?(stage)
     return false unless stage.lockable?
     return false if user.authorized_teacher?
-    submitted? && !readonly_answers? || has_autolocked?(stage)
+    locked
+  end
+
+  # `readonly` and `locked` are mutually exclusive on the client, so we use
+  # this helper to override the value of `readonly_answers` when we're supposed
+  # to show as locked.
+  def show_as_readonly?(stage)
```

Comment:
Does it help readability if we call this `show_as_readonly_answers` or `show_answers`?  (If `show_as_readonly_answers` is too long, I think the "answers" part is more important than the "readonly" part?)

Reply:
If I'm following the below conversation correctly, it sounds like we've all come to the consensus to leave this as-named for now, and we can come back around to rethinking/renaming in future work.